v5/core/core.go

```diff
@@ -105,10 +106,22 @@ func (core *Core) Setup(sdkSettings model.SdkSettings, deviceProperties model.De
 		core.environment = client.NewSaasEnvironment(envApi)
 	}
 
-	// TODO Analytics.Analytics.Initialize(deviceProperties.RolloutKey, deviceProperties)
-
 	core.internalFlags = client.NewInternalFlags(core.experimentRepository, core.parser, core.environment)
-	core.impressionInvoker = impression.NewImpressionInvoker(core.internalFlags, core.customPropertyRepository, deviceProperties /* TODO Analytics.Analytics.Client, */, roxyPath != "")
+	analyticsHandler := analytics.NewAnalyticsHandler(&analytics.AnalyticsDeps{
+		UriPath:           core.environment.EnvironmentAnalyticsPath(),
+		Request:           network.NewRequest(http.DefaultClient),
+		DeviceProperities: deviceProperties,
```

Comment:
Most recent commit moved disablement to core setup. Was pointless to create and pass analytics handler when analytics is disabled at the core.
Especially because core.Shutdown needs direct access to stop the analytics interval reporting.

The other properties are now exposed.

With the exception of the logger. The logger interface is simply for testing. If no logger is passed, then the analyticsHandler setup function (new) will set the logger.